doc/read-the-docs-site/doc/marconi-as-a-library/tutorials/BasicApp.hs

```diff
@@ -0,0 +1,484 @@
+import Cardano.BM.Setup qualified as Trace
+import Cardano.BM.Tracing qualified as Trace
+import Marconi.ChainIndex.Experimental.Indexers.Orphans ()
+import Marconi.ChainIndex.Experimental.Indexers.SyncHelper qualified as Core
+import Marconi.ChainIndex.Experimental.Runner qualified as Core
+import Marconi.ChainIndex.Node.Client.Retry qualified as Core
+import Marconi.ChainIndex.Orphans ()
+import Marconi.ChainIndex.Types (BlockEvent (BlockEvent), SecurityParam)
+import Marconi.ChainIndex.Utils qualified as Utils
+import Marconi.Core qualified as Core
+import Marconi.Core.JsonRpc qualified as Core
+import Network.JsonRpc.Types (JsonRpc, RawJsonRpc)
+
+-- BLOCKEND import
+
+main :: IO ()
+main = runBlockInfoSqliteIndexerHttp
+
+-- BLOCKSTART env
+data Env indexer = Env
+  { envTrace :: !(Trace.Trace IO Text)
+  -- ^ The 'Trace' used for logging
+  , envSocketFilePath :: !String
+  -- ^ The local file path of the local node's socket file
+  , envNetworkId :: !C.NetworkId
+  -- ^ Network Id of the local node
+  , envSecurityParam :: !SecurityParam
+  -- ^ Security parameter (K) of the local node
+  , envIndexerWorker :: !(Core.WorkerIndexer IO BlockEvent BlockInfoEvent indexer)
+  -- ^ Worker of the indexer we want to run.
+  --
+  -- * 'BlockEvent': the data we get from the local node.
+  -- * 'BlockInfoEvent': the event type
+  -- * 'indexer' the indexer backend (ListIndexer, SQLiteIndexer, FileIndexer, etc.)
+  }
+
+withIndexerBuildEnv
+  :: Core.WorkerIndexer IO BlockEvent BlockInfoEvent indexer
+  -> (Env indexer -> IO ())
+  -> IO ()
+withIndexerBuildEnv worker action = do
+  -- We start by getting the network id and socket file path of the local node.
+  -- We get the values through the CLI.
+  [socketFilePath, networkMagicStr] <- getArgs
+  let networkMagic = read networkMagicStr
+
+  -- The security parameter.
+  -- TODO Should be queried from the local node.
+  let k = 2_160
+
+  -- We initialise the Trace for logging from `iohk-monitoring.
+  -- We just use default values.
+  traceConfig <- Trace.defaultConfigStdout
+  Trace.withTrace traceConfig "marconi-tutorial" $ \stdoutTrace -> do
+    let networkId = C.Testnet $ C.NetworkMagic networkMagic
+    action (Env stdoutTrace socketFilePath networkId k worker)
+
+-- BLOCKEND env
+
+-- BLOCKSTART runBlockInfoListIndexer
+runBlockInfoListIndexer :: IO ()
+runBlockInfoListIndexer = do
+  worker <- mkBlockInfoListIndexerWorker
+  withIndexerBuildEnv worker $ \env -> liftIO $ runIndexer env
+
+-- BLOCKEND runBlockInfoListIndexer
+
+-- BLOCKSTART runBlockInfoSqliteIndexer
+runBlockInfoSqliteIndexer :: IO ()
+runBlockInfoSqliteIndexer = do
+  worker <- mkBlockInfoSqliteIndexerWorker ":memory:"
+  withIndexerBuildEnv worker $ \env -> liftIO $ runIndexer env
+
+-- BLOCKEND runBlockInfoSqliteIndexer
+
+-- BLOCKSTART runBlockInfoIndexer
+runIndexer :: Env indexer -> IO ()
+runIndexer env = do
+  -- TODO Why do we need a coordinator here? We should be able to start a worker with `runIndexer`.
```

Comment:
This TODO is a bit opaque to newcomers. Can you elaborate?

Reply:
Changed it. Can you revalidate? https://github.com/input-output-hk/marconi/pull/202/files#diff-e92ed9cee78c396f00fdc33f17b96888edd487292f32bb3ebd9bcee50062eb3fR123